mitodl/open-discussions
# moderator notification setting ui

#### Pre-Flight checklist

<img width="1220" alt="Screen Shot 2020-12-14 at 7 59 09 AM" src="https://user-images.githubusercontent.com/1934992/102085487-b4cb6b00-3de4-11eb-9a1e-798f0d28b125.png">

#### What are the relevant tickets?
closes https://github.com/mitodl/open-discussions/issues/3244

#### What's this PR do?
This PR adds an option in /settings/notifications that is only visible to moderators to allow moderators to individually opt in or out of notifications

#### How should this be manually tested?
Run `docker-compose run web ./manage.py populate_notification_settings`

Log in as a non-moderator user and go to `http://localhost:8063/settings/`. There should be no changes

Log in as a moderator user and go to `http://localhost:8063/settings/`. There should be a new checkbox to allow moderator notifications.

Keep moderator notifications enabled in that ui and enable notifications in `http://localhost:8063/manage/c/edit/<channel>/basic/`

Make a post in the channel. Then run

```
from notifications.tasks import *
send_unsent_email_notifications()
```
You should get a notification on the MAILGUN_RECIPIENT_OVERRIDE email

If moderator notifications are turned off for either the user or the channel, you should not receive a notification email

### Updated you based on @Ferdi's comment:
<img width="1215" alt="Screen Shot 2020-12-22 at 2 11 34 PM" src="https://user-images.githubusercontent.com/1934992/102924593-e5a04580-445f-11eb-9003-16455538136d.png">

### Updated UI based on additional comment from @Ferdi:
<img width="1333" alt="Screen Shot 2021-01-11 at 9 02 17 PM" src="https://user-images.githubusercontent.com/1934992/104259966-8034e880-5450-11eb-9698-10dd4215741e.png">
